src/components/GraphiQL.js

```diff
@@ -167,7 +167,12 @@ export class GraphiQL extends React.Component {
     if (nextSchema !== this.state.schema ||
         nextQuery !== this.state.query ||
         nextOperationName !== this.state.operationName) {
-      this._updateQueryFacts(nextQuery);
+      const newOperationName = this._updateQueryFacts(nextQuery);
```

Comment:
Notice that schema can also differ from what is in `this.state` which will also cause a race condition. `schema` should also be provided as an argument to `_updateQueryFacts` to avoid this.